storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/BytesColumnVector.java

```diff
@@ -213,18 +211,17 @@ public void setVal(int elementNum, byte[] sourceBuf) {
    * Ensures that we have space allocated for the next value, which has size
    * length bytes.
    *
-   * Updates currentValue, currentOffset, and sharedBufferOffset for this value.
+   * Updates currentValue and currentOffset for this value.
    *
-   * Always use before getValPreallocatedBytes, getValPreallocatedStart,
-   * and setValPreallocated.
+   * Always use before getValPreallocatedBytes, getValPreallocatedStart.
+   * setValPreallocated must be called to actually reserve the bytes.
    */
   public void ensureValPreallocated(int length) {
     if ((sharedBufferOffset + length) > sharedBuffer.length) {
       currentValue = allocateBuffer(length);
```

Comment:
This is unrelated to the fix but feels like only currentValue is getting assigned while currentOffset is also being changed inside allocateBuffer method.

Since allocateBuffer is private and not used elsewhere, we could move both the assignments of currentValue and currentOffset into the allocateBuffer method and have this as

```java
    if ((sharedBufferOffset + length) > sharedBuffer.length) {
      allocateBuffer(length);
    } else {
      currentValue = sharedBuffer;
      currentOffset = sharedBufferOffset;
    }
```